rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -229,6 +229,30 @@ void Player::play()
   std::lock_guard<std::mutex> lk(ready_to_play_from_queue_mutex_);
   is_ready_to_play_from_queue_ = false;
   ready_to_play_from_queue_cv_.notify_all();
+
+  // Wait for all published messages to be acknowledged.
+  if (play_options_.wait_acked_timeout >= 0) {
+    std::chrono::milliseconds timeout(play_options_.wait_acked_timeout);
+    if (timeout == std::chrono::milliseconds(0)) {
+      timeout = std::chrono::milliseconds(-1);
+    }
+    for (auto pub : publishers_) {
+      try {
+        if (!pub.second->wait_for_all_acked(timeout)) {
+          RCLCPP_ERROR(
+            get_logger(),
+            "Timeout to wait all published messages acknowledged for topic %s",
```

Comment:
I feel like this reads a little better.

```suggestion
            "Timed out while waiting for all published messages to be acknowledged for topic %s",
```